www/docs/server/procedures.md

```diff
@@ -5,336 +5,85 @@ sidebar_label: Define Procedures
 slug: /server/procedures
 ---
 
-Procedures in tRPC are very flexible primitives to create backend functions; they use a builder pattern which means you can create reusable base procedures for different parts of your backend application.
+In tRPC a Procedure is ultimately just a function which you want to expose to your client. A Procedure can be either:
+
+- a `Query` - used to fetch data, generally does not change any data
+- a `Mutation` - used to send data, often for create/update/delete purposes
+
+Procedures in tRPC are very flexible primitives to create backend functions, they use an immutable builder pattern which means you can [create reusable base procedures](#reusable-base-procedures) which share functionality among multiple procedures.
 
 :::tip
 
-- A procedure can be viewed as the equivalent of a REST-endpoint or a function.
+- A procedure can be viewed as the equivalent of a REST-endpoint, and under the hood are built on HTTP, but are more focused towards the needs of application, rather than representing "resources".
 - There's no internal difference between queries and mutations apart from semantics.
-- Defining publicProcedure is the same for queries, mutations, and subscription with the exception that subscriptions need to return an `observable` instance.
+- Defining a publicProcedure is the same for queries, mutations, and subscriptions, with the exception that subscriptions need to return an `observable` instance.
 
 :::
 
-## Example without input validation
+## Writing procedures
+
+Writing a procedure is simple, the `t` object you create during tRPC setup returns an initial `t.procedure` which all other procedures are built on:
```

Comment:
might be good to explain why we are adding the middlestep of `publicProcedure` ?

Reply:
I've added a note now in the base-procedures bit. Though if there's more to it please feel free to suggest a change, I might not have all the context but just assume it's to give a meaningful name for export when read alongside other base procedures :)